code/src/cljs/sixsq/nuvla/ui/edge_detail/views.cljs

```diff
@@ -211,7 +212,8 @@
     (swap! form-data assoc :environment (str/join "\n" environment))
     (fn [{:keys [id] :as _resource} _operation show? title icon button-text]
       (let [correct-nb? (= (:parent @status) id)
-            nb-version  (get @status :nuvlabox-engine-version "")]
+            nb-version  (get @status :nuvlabox-engine-version "")
+            target-version  (:key (get (zipmap (map :value @releases) @releases) (:nuvlabox-release @form-data)))]
```

Comment:
You can do easier than this line:
(:key (get (zipmap (map :value @releases) @releases) (:nuvlabox-release @form-data)))
vs
(->> @releases
        (some #(when (= (:value %) (:nuvlabox-release @form-data)) %))
         :key))